content/tutorials/MQTT-snake-js.textile

```diff
@@ -0,0 +1,255 @@
+
+<%= partial 'tutorials/_step-1-setup-free-account' %>
+
+h2. Step 2 - Set up your device
+
+Now that you have an Ably account and have your API Key, you can work on setting up your controller. Firstly, you'll need to create a folder and enter it. To do this type the following into your command line:
+
+```[sh]
+  mkdir mqtt-snake
+  cd mqtt-snake
+```
+
+You'll also need the MQTT.js and keypress NPM modules for the controller, so type the following into your command line to get them:
+
+```[sh]
+  npm install mqtt --save
+  npm install keypress —-save
+```
+
+You'll now see all the required files inside the @node_modules@ folder that has been created within the @mqtt-snake@ folder.
+
+"MQTT.js":https://www.npmjs.com/package/mqtt is required for you to use MQTT in Javascript, and "keypress":https://www.npmjs.com/package/keypress allows for you to easily detect keys being pressed on the keyboard.
+
+"See this step in Github":https://github.com/ably/tutorials/TODO
+
+h2. Step 3 - Add your libraries
+
+With your NPM modules ready, it's time to create a file to contain your controller code. Create a file called @controller.js@ inside your @mqtt-snake@ folder. Inside this file, instance the MQTT.js and keypress libraries:
+
+```[javascript]
+  const mqtt = require('mqtt');
+  const keypress = require('keypress');
+```
+
+"See this step in Github":https://github.com/ably/tutorials/TODO
+
+h2. Step 4 - Detect key presses
+
+Now it's time for you to detect key presses on the keyboard. In this tutorial, you'll be needing to know when the arrow keys or return key is pressed, and then through MQTT communicate which key was pressed. You'll be detecting the key presses with the keypress NPM module. Add the below code after the library instancing in the @controller.js@ file:
+
+```[javascript]
+  keypress(process.stdin);
+
+  process.stdin.setRawMode(true);
+
+  process.stdin.on('keypress', function (ch, key) {
+    if (key) {
+      console.log('Key clicked: ' +  key.name);
+    }
+  });
+```
+
+There are three main things occuring in this code. The first is @keypress(process.stdin)@, which will simply cause our @process.stdin@ to emit @keypress@ events.
+
+Secondly is @process.stdin.setRawMode(true)@, which ensures your keyboard input is available character-by-character with no modifiers. For example, when typing ctrl-c, @process.stdin@ will output the following as a @keypress@ event: 
+
+```
+  { name: 'c',
+    ctrl: true,
+    meta: false,
+    shift: false,
+    sequence: '\u0003' 
+  }
+```
+
+Finlly, @process.stdin.on()@ is simply a listener for the @keypress@ event. You use it in this case to print to the console whenever a key is pressed. You can try this out for yourself by running the following line of code in the terminal:
+
+```[sh]
+  node controller.js
+```
+
+If all is well, you should see a message saying the key you pressed whenever you click a key.
```

Comment:
Maybe clearer to say ' you should see a message reflecting the key you pressed, whenever'